check and display form validations

joshkendrick opened the pull request:
This adds running the validations before form submission, and should appropriately show error messages below fields. This probably completes [UMC-41](https://issues.boundlessgeo.com:8443/browse/UMC-41).

`required`, numeric `minimum` and `maximum` seem to be working correctly. All other value constraints need to be added for [UMC-42](https://issues.boundlessgeo.com:8443/browse/UMC-42).

joshkendrick committed:
check and display form validations

joshkendrick committed:
copy the config for text based fields

joshkendrick committed:
always copy the field schema into config

this is so the field schema is available for other things, for example, validation

joshkendrick committed:
fix and finish validation checks

frankrowe commented on src/SCForm.native.js:
```diff
@@ -79,14 +80,15 @@ class SCForm extends Component {
-        { text: 'Cancel' },
-        { text: 'Submit', onPress: () => this.props.saveForm(formData) },
-      ]);
+    if (validateFields(this.form)) {
```
Instead of passing the form ref here, I think we should pass the form value, and `this.state.options`. Then we can create a new options object with any new errors and return it. Then back in the component we could do `this.setState({options: newOptions})`. This way we don't have to mutate the form ref or call `forceUpdate`.

frankrowe commented on src/validateFields.js:
```diff
@@ -0,0 +1,93 @@
+    options.error = errors;
+    // then we must force an update if options.hasError is already true
+    if (field.hasError()) {
+      field.forceUpdate();
```
try to avoid `forceUpdate`. the component should  automatically update when receiving new state or props.

joshkendrick replied:
yeah i couldnt find a way to make this work. it seemed that react-native wouldn't re-draw the field if i just changed the `error`. it would if you changed `hasError` because i think that's part of the state, but `error` isnt. the situation was when they were below the minimum, then change the value to be above the maximum, the error message didnt update.

joshkendrick committed:
pass form values and schema to validate

this should be a more correct way of validating fields, to jive with code review comments

joshkendrick commented:
@frankrowe if you could check this again when you get time, that would be 👌

frankrowe commented:
@joshkendrick new changes looking good 👍
